tests/unit/test_metrics.c

```diff
@@ -0,0 +1,78 @@
+﻿#include "sentry_metrics.h"
+#include "sentry_testsupport.h"
+
+SENTRY_TEST(metrics_name_sanitize)
+{
+    char *name1 = sentry__metrics_sanitize_name("foo-bar");
+    char *name2 = sentry__metrics_sanitize_name("foo\\!,bar");
+    char *name3 = sentry__metrics_sanitize_name("fo%-bar");
+    // char *name4 = sentry__metrics_sanitize_name("fôo-bär");
+
+    TEST_CHECK_STRING_EQUAL(name1, "foo-bar");
+    TEST_CHECK_STRING_EQUAL(name2, "foo\\__bar");
+    TEST_CHECK_STRING_EQUAL(name3, "fo_-bar");
+    // TEST_CHECK_STRING_EQUAL(name4, "f_o-b_r");
+
+    sentry_free(name1);
+    sentry_free(name2);
+    sentry_free(name3);
+}
+
```

Comment:
I strongly recommend using a convenience macro so you don't have to track frees for every test case. One could argue that you won't run out of memory if you forget to free, but a more significant argument would be that it makes the test-cases much easier to follow (and to extend).

Something like

```C
#define TEST_CHECK_METRICS_SANITY(fn, input, expected)                         \
    {                                                                          \
        char *actual = fn(input);                                              \
        TEST_CHECK_STRING_EQUAL(actual, expected);                             \
        sentry_free(actual);                                                   \
    }
```

Allows you to write the test above like this

```C
    TEST_CHECK_METRICS_SANITY(
        sentry__metrics_sanitize_name, "foo-bar", "foo-bar");
    TEST_CHECK_METRICS_SANITY(
        sentry__metrics_sanitize_name, "foo\\!,bar", "foo\\__bar");
    TEST_CHECK_METRICS_SANITY(
        sentry__metrics_sanitize_name, "fo%-bar", "fo_-bar");
    TEST_CHECK_METRICS_SANITY(
        sentry__metrics_sanitize_name, "fôo-bär", "f_o-b_r");
    TEST_CHECK_METRICS_SANITY(
        sentry__metrics_sanitize_name, "f👾o✓-br🎨€", "f_o_-br__");
```

Reply:
Thank you for sharing this - the tests look much cleaner with the above macro compared to the current version.